lib/fizzy/capi.cpp

```diff
@@ -7,10 +7,79 @@
 #include "instantiate.hpp"
 #include "parser.hpp"
 #include <fizzy/fizzy.h>
+#include <cstring>
 #include <memory>
 
 namespace
 {
+inline void set_success(FizzyError* error) noexcept
+{
+    if (error == nullptr)
+        return;
+
+    error->code = FIZZY_SUCCESS;
+    error->message[0] = '\0';
+}
+
+template <size_t DestSize>
```

Comment:
Could the template work with a fixed-sized array for src? i.e. `src[N]` and using N as DestSize?